pkg/web/prometheus.go

```diff
@@ -0,0 +1,66 @@
+package web
+
+import (
+	"fmt"
+	"net/http"
+
+	"log"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// TODO: now I have setup the Prometheus metric objects, how do I integrate with gostatsd internal metrics measuring
+// tools? How do I run this file, and start writing some tests for this. Need to adhere to the project testing styles.
```

Comment:
The expected interface is a `stats.Statser` (https://github.com/atlassian/gostatsd/blob/master/pkg/stats/statser.go), and you can embed a `stats.flushNotifier` to take care of the `RegisterFlush` and `NotifyFlush` calls.

You'll likely need to lazily create metrics on-demand and store them in a map - gostatsd doesn't have any arbitrary metrics, so you don't need to worry about removing things from the map.  The map can be protected with an RWMutex, since after warmup, it will be very rare (if ever), that additional metrics are created.

Reply:
Also, there's basically no testing of the actual Statsers.  There is a bit for `pkg/web`, but it's acceptable to just make sure you get a 200 return code when it's expected.